Formula/sftpgo.rb

```diff
@@ -0,0 +1,74 @@
+class Sftpgo < Formula
+  desc "Fully featured SFTP server with optional HTTP/S, FTP/S and WebDAV support"
+  homepage "https://github.com/drakkan/sftpgo"
+  url "https://github.com/drakkan/sftpgo/releases/download/v2.2.3/sftpgo_v2.2.3_src_with_deps.tar.xz"
+  sha256 "6c8676725e86ee3f6ad46a340a84f0da37cab8b6ea7b6aee86b2b96ba5e6671a"
+  license "AGPL-3.0-only"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w")
+    system bin/"sftpgo", "gen", "man", "-d", "man"
+    man1.install Dir["man/*.1"]
+
+    (zsh_completion/"_sftpgo").write Utils.safe_popen_read(bin/"sftpgo", "gen", "completion", "zsh")
+    (bash_completion/"sftpgo").write Utils.safe_popen_read(bin/"sftpgo", "gen", "completion", "bash")
+    (fish_completion/"sftpgo.fish").write Utils.safe_popen_read(bin/"sftpgo", "gen", "completion", "fish")
+
+    inreplace "sftpgo.json" do |s|
+      s.gsub! "\"users_base_dir\": \"\"", "\"users_base_dir\": \"#{var}/sftpgo/data\""
+      s.gsub! "\"backups_path\": \"backups\"", "\"backups_path\": \"#{var}/sftpgo/backups\""
+      s.gsub! "\"credentials_path\": \"credentials\"", "\"credentials_path\": \"#{var}/sftpgo/credentials\""
+      s.gsub! "\"templates_path\": \"templates\"", "\"templates_path\": \"#{opt_pkgshare}/templates\""
+      s.gsub! "\"static_files_path\": \"static\"", "\"static_files_path\": \"#{opt_pkgshare}/static\""
+      s.gsub! "\"openapi_path\": \"openapi\"", "\"openapi_path\": \"#{opt_pkgshare}/openapi\""
+    end
```

Comment:
Could you make an upstream issue to make these configurable at build time? It doesn't have to block merging, but this isn't the most future proof way to set these

Reply:
> Could you make an upstream issue to make these configurable at build time? It doesn't have to block merging, but this isn't the most future proof way to set these

@SMillerDev , I'm the upstream :smile:.

I removed some replaces. In the future I can add some automatic search paths for `templates_path`, `static_files_path` and `static_files_path`. The only one that must stay is the `users_base_dir`. This dir defines the global default path for user's home dir (where the user data is stored) and so end users should be able to change it.

I'll add these changes for v2.3.0 (hopefully no other 2.2.x releases will happen).

I'm concerned regarding the user running the service, based on the info [here](https://www.launchd.info/), a global daemon runs on behalf of root or the user specified with the key `UserName`. So I cannot have the SFTPGo service to automatically start on boot as non-root user without defining a custom plist.

Reply:
> Could you make an upstream issue to make these configurable at build time? It doesn't have to block merging, but this isn't the most future proof way to set these

the directories in the shared data paths are now automatically discovered

https://github.com/drakkan/sftpgo/commit/80da2dc722ff50f64a73ee5c1955bfd8de89dc27#diff-a61d5e39f45fb5ed2dc1ef71fa708c8db480e9a396adb8a5d9d17b8e6ba1e186R352

so in v2.3.0 we can remove the replace for `static`, `templates` and `openapi` from the formula, only `users_base_dir` will remain